tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -11299,6 +11299,62 @@ bool Sema::DiagnoseHLSLMethodCall(const CXXMethodDecl *MD, SourceLocation Loc) {
   return false;
 }
 
+void ValidateWaveSize(clang::Sema *S, FunctionDecl *FD) {
+  const auto *SM =
+      hlsl::ShaderModel::GetByName(S->getLangOpts().HLSLProfile.c_str());
+
+  const llvm::StringRef &entryName = S->getLangOpts().HLSLEntryFunction;
+  StringRef functionName = "";
+  if (FD->getIdentifier())
+    functionName = FD->getName();
+  bool isEntry = false;
+  bool isNode = false;
+  bool isCS = false;
+
+  isEntry =
+      !SM->IsLib() &&
```

Comment:
A more convenient way to perform this check in this environment would be to use the boolean that has already been assigned in the langopts : `self->getLangOpts().IsHLSLLibrary`. No (further) string parsing required.